File: support_tools/swigwin-3.0.12/Examples/test-suite/python/director_extend_runme.py
# Test case from bug #1506850
#"When threading is enabled, the interpreter will infinitely wait on a mutex the second
# time this type of extended method is called. Attached is an example
# program that waits on the mutex to be unlocked."

from director_extend import *


class MyObject(SpObject):

    def __init__(self):
        SpObject.__init__(self)
        return

    def getFoo(self):
        return 123

m = MyObject()
if m.dummy() != 666:
    raise RuntimeError, "1st call"
if m.dummy() != 666:                        # Locked system
    raise RuntimeError, "2nd call"

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/director_extend_runme.go
// Test case from bug #1506850 "When threading is enabled, the
// interpreter will infinitely wait on a mutex the second time this
// type of extended method is called.  Attached is an example program
// that waits on the mutex to be unlocked."

package main

import . "./director_extend"

func main() {
	m := NewSpObject()
	if m.Dummy() != 666 {
		panic("1st call")
	}
	if m.Dummy() != 666 {
		panic("2nd call")
	}
}
